osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -632,7 +632,7 @@ private void updateItems()
                         case DrawableCarouselBeatmap beatmap:
                         {
                             if (beatmap.Item.State.Value == CarouselItemState.Selected)
-                                scrollTarget = currentY + beatmap.DrawHeight / 2 - DrawHeight / 2;
+                                scrollTarget = currentY + beatmap.DrawHeight / 2 - (Parent.DrawHeight / 2 - Parent.Padding.Top);
```

Comment:
While this change seems correct it took me about half an hour to work out what this exactly did. The double negation especially threw me in for a loop. I'd say this should read
```suggestion
                                scrollTarget = currentY - Parent.DrawHeight / 2 + Parent.Padding.Top + beatmap.DrawHeight / 2;
```
and potentially have a comment explaining what the last part is (first subtracting half of height to end up with the element at the middle of the screen, and then adding back possible top padding from parent and then half of our own height to achieve vertical centering of the item itself).

Reply:
Yeah I drew a graph for myself with like 6 different variables and arrived at the correct solution via substitution into one of the equations. Should have included that in the PR description at the very least, maybe even in code as a comment.

Reply:
I made my crude drawing into following image, should this be included as a comment?
![carouselDrawLogic](https://user-images.githubusercontent.com/11320483/79073699-b04c3100-7cf0-11ea-88fd-e0e8196a517b.png)

Reply:
The image is nice but I'd probably leave it as part of this PR and have a textual comment in the source.

Reply:
Is having a link to PR in the comment fine?

Reply:
I'd much prefer just plain text, no links. Links are prone to breakage. (You could say that if github goes away we have bigger problems but IMO source should be stand-alone as much as possible.)

Reply:
Could I ask you to write that comment, like edit your original suggestion? I can't actually figure out a nice way to comment that in plain text.

Reply:
Maybe something like
```csharp
// scroll position at currentY makes the set panel appear at the very top of the carousel in screen space
// move down by half of parent height (which is the height of the carousel's visible extent, including semi-transparent areas)
// then reapply parent's padding from the top by adding it
// and finally add half of the panel's own height to achieve vertical centering of the panel itself
```